### DISCUSSION TITLE
TSDoc and TypeScript support

### DISCUSSION THREAD

**Discussion by RunDevelopment** [hosseinmd/prettier-plugin-jsdoc#68] [Ideas]
This is something I wondered. This plugin does format TypeScript files but does it support TSDoc? (A: No.)

TSDoc is [supposed to be similar to JSDoc](https://tsdoc.org/#:~:text=Familiar%20syntax) but it's either a subset or a superset. E.g. TSDoc treats the [`@example` tag](https://tsdoc.org/pages/tags/example/) very differently from JSDoc. There are also a bunch of other minor differences.

So I have two questions:

1. Is this plugin supposed to support TSDoc?
2. Should this plugin really format TS files?
    (I know that it's useful but it is going to mess up the documentation of some.)

Or in one question: What is the state of TSDoc and TypeScript support of this plugin?

**Comment by fardad-dev** [comment]
We want support typescript full.

**Reply by RunDevelopment** [reply]
I will take that as a "yes" to my 2. question.

That's good. I want to use this plugin in my TS projects so I'm happy to hear that.

**Reply by RunDevelopment** [reply]
So can I expect that TSDoc will be fully supported one day?

**Reply by fardad-dev** [reply]
Yes, but I don't tested that.
